community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java

```diff
@@ -131,7 +131,7 @@ TransactionType upgradeToSchemaTransaction() throws InvalidTransactionTypeKernel
     private TransactionHooks.TransactionHooksState hooksState;
     private KernelStatement currentStatement;
```

Comment:
`KernelTransactionImplementation` still contains `TransactionType` enumeration and one of possible state is READ which intersects with AccessMode values and actual responsibility.